src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/UnusedTagsPass.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Find all service tags which are defined, but not used and yield a warning log message.
+ *
+ * @author Florian Pfitzer <pfitzer@wurzel3.de>
+ */
+class UnusedTagsPass implements CompilerPassInterface
+{
+    /**
+     * whitelisted tags
+     *
+     * @var array
+     */
+    protected $whitelist = array(
```

Comment:
I wouldn't make it protected by default. If we want to offer an extension point, I'd recommend a separate method to add a tag to the whitelist.